include/dslash_helper.cuh

```diff
@@ -429,6 +491,129 @@ namespace quda
       // first few blocks do packing kernel
       P<dagger, dslash.pc_type(), Arg> packer;
       packer(arg, s, 1 - parity, dslash.twist_pack()); // flip parity since pack is on input
+
+      // we use that when running the exterior -- this is either
+      // * an explicit call to the exterior when not merged with the interior or
+      // * the interior with ext_blocks > 0
+#ifdef NVSHMEM_COMMS
+    } else if (arg.shmem > 0
```

Comment:
Can we move the contents of this to a separate function to make `dslashGPU` more readable from a high level?